vm/src/stdlib/itertools.rs

```diff
@@ -953,6 +953,87 @@ impl PyItertoolsCombinations {
     }
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct PyItertoolsZiplongest {
+    iterators: Vec<PyObjectRef>,
+    fillvalue: PyObjectRef,
+    numactive: RefCell<usize>,
+}
+
+impl PyValue for PyItertoolsZiplongest {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "zip_longest")
+    }
+}
+
+#[derive(FromArgs)]
+struct ZiplongestArgs {
+    #[pyarg(keyword_only, optional = true)]
+    fillvalue: OptionalArg<PyObjectRef>,
+}
+
+#[pyimpl]
+impl PyItertoolsZiplongest {
+    #[pyslot(new)]
+    fn tp_new(
+        cls: PyClassRef,
+        iterables: Args,
+        args: ZiplongestArgs,
+        vm: &VirtualMachine,
+    ) -> PyResult<PyRef<Self>> {
+        let fillvalue = match args.fillvalue.into_option() {
+            Some(i) => i,
+            None => vm.get_none(),
+        };
+
+        let iterators = iterables
+            .into_iter()
+            .map(|iterable| get_iter(vm, &iterable))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let numactive = RefCell::new(iterators.len());
+
+        PyItertoolsZiplongest {
+            iterators,
+            fillvalue,
+            numactive,
+        }
+        .into_ref_with_type(vm, cls)
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        if self.iterators.is_empty() {
+            Err(new_stop_iteration(vm))
+        } else {
+            let mut next_obj: PyObjectRef;
+            let mut result: Vec<PyObjectRef> = Vec::new();
+            let mut numactive = self.numactive.clone().into_inner();
+
+            for idx in 0..self.iterators.len() {
+                next_obj = match call_next(vm, &self.iterators[idx]) {
+                    Ok(obj) => obj,
+                    Err(_) => {
```

Comment:
You might want to check if the error here is a `StopIteration` or not and handle it accordingly (if it isn't `StopIteration` we just want to pass it on):
```rust
Err(err) => {
    if !objtype::isinstance(&err, &vm.ctx.exceptions.stop_iteration) {
        return Err(err);
    }
    numactive -= 1;
    // ...
}
```